### PR TITLE
CDRW-4343 - Add CategoryPurposeCode

### PR DESCRIPTION
**Pull request by charlieGreenyOB**
added to:
- OBRisk1 (Payments)
- OBReadTransaction6/Data/Transaction

### CODE REVIEW

File: dist/openapi/account-info-openapi.yaml [hunk 1]
```diff
@@ -5049,6 +5049,56 @@ components:
           $ref: '#/components/schemas/Name_0'
         SecondaryIdentification:
           $ref: '#/components/schemas/SecondaryIdentification'
+    OBCategoryPurposeCode:
+      description: >-
+        Enumeration of codes that outlines the type of purpose behind a
+        transaction, payment or risk.
```
Comment (wkrop) [inline dist/openapi/account-info-openapi.yaml:5055]: No need for breaking the string over 2 lines

File: dist/openapi/payment-initiation-openapi.json [hunk 2]
```diff
@@ -4973,6 +4973,14 @@
         "minLength": 1,
         "maxLength": 16
       },
+      "OBCategoryPurposeCode": {
+        "description" : "Enumeration of codes that outlines the type of purpose behind a transaction, payment or risk.",
+        "type" : "string",
+        "enum" : ["BONU","CASH","CBLK","CCRD","CGWV","CIPC","CONC","CORT","DCRD","DIVI","DVPM","EPAY","FCDT",
+        "FCIN","FCOL","GOVT","GP2P","HEDG","ICCP","IDCP","INTC","INTE","LBOX","LOAN","MP2B","MP2P","OTHR",
+        "PENS","RPRE","RRCT","RVPM","SALA","SECU","SSBE","SUPP","SWEP","TAXS","TOPG","TRAD","TREA","VATX",
+        "VOST","WHLD","ZABA"]
```
Comment (wkrop) [inline dist/openapi/payment-initiation-openapi.json:4982]: Enum array values should be new line separated in JSON files.

File: dist/openapi/payment-initiation-openapi.yaml [hunk 3]
```diff
@@ -3047,6 +3047,56 @@ components:
       type: string
       minLength: 1
       maxLength: 16
+    OBCategoryPurposeCode:
+      description: >-
+        Enumeration of codes that outlines the type of purpose behind a
+        transaction, payment or risk.
```
Comment (wkrop) [inline dist/openapi/payment-initiation-openapi.yaml:3053]: No need for breaking the string over 2 lines

File: dist/openapi/vrp-openapi.json [hunk 4]
```diff
@@ -1834,6 +1834,14 @@
           }
         }
       },
+      "OBCategoryPurposeCode": {
+        "description" : "Enumeration of codes that outlines the type of purpose behind a transaction, payment or risk.",
+        "type" : "string",
+        "enum" : ["BONU","CASH","CBLK","CCRD","CGWV","CIPC","CONC","CORT","DCRD","DIVI","DVPM","EPAY","FCDT",
+        "FCIN","FCOL","GOVT","GP2P","HEDG","ICCP","IDCP","INTC","INTE","LBOX","LOAN","MP2B","MP2P","OTHR",
+        "PENS","RPRE","RRCT","RVPM","SALA","SECU","SSBE","SUPP","SWEP","TAXS","TOPG","TRAD","TREA","VATX",
+        "VOST","WHLD","ZABA"]
```
Comment (wkrop) [inline dist/openapi/vrp-openapi.json:1843]: Enum array values should be new line separated in JSON files.

**Review by wkrop — CHANGES_REQUESTED** [review]

File: dist/openapi/payment-initiation-openapi.yaml (hunk 3 above)
Comment (charlieGreenyOB) [reply dist/openapi/payment-initiation-openapi.yaml:3053]: This was generated from converting the relevant JSON, is it okay to leave this?

**Review by wkrop — APPROVED** [review]

**Pull request merged**